loki/transformations/utilities.py

```diff
@@ -623,20 +639,94 @@ def find_driver_loops(section, targets):
         List of subroutines that are to be considered as part of
         the transformation call tree.
     """
+    class FindDriverLoops(Visitor):
```

Comment:
Since the class is defined insided the routine, you don't have to pass the `targets` value as an argument but can directly use it. Best to make sure it's an iterable first, though:
```suggestion
    targets = [str(t).lower() for t in as_tuple(targets)]

    class FindDriverLoops(Visitor):
```